Remove DUNE_PROFILE variable in configure

Halbaroth opened the pull request:
This is not the appropriate way to ensure that dune runs with release profile. The `DUNE_PROFILE` environment variable is the right way to invoke the Makefile with a specific profile. For instance,
```console
DUNE_PROFILE=release make build
```
builds the project with release profile.

Halbaroth committed:
Remove DUNE_PROFILE variable in configure

This is not the appropriate way to ensure that dune runs with release
profile. The `DUNE_PROFILE` environment variable is the right way to
invoke the Makefile with a specific profile. For instance,
```console
DUNE_PROFILE=release make build
```

Halbaroth commented:
I also fixed an issue with the ancient option in `configure.ml`. The project was compiled with Ancient as soon as it was installed in the switch.

a2line commented:
The changes broke test metrics collection and Windows builds, but I got [CI passing on Windows](https://github.com/a2line/geneweb/actions/runs/15404538227) by changing `ci` target from `dune runtest` to `dune build @runtest`, allowing dune to intelligently build only the dependencies needed for tests instead of requiring a full `build` dependency. I also corrected the step references in the workflow from `steps.build.outputs` to `steps.tests.outputs` since test metrics are now collected in the new separate "Run tests" step with a new id "tests".

```diff
# Makefile change - dune handle test dependencies intelligently
-ci: | $(GENERATED_FILES_DEP) build
-	@GENEWEB_CI=on dune runtest
+ci: | $(GENERATED_FILES_DEP)
+	@GENEWEB_CI=on dune build @runtest

# Workflow fix - reference correct step for test metrics (l. 176-179)
-          tests-total: ${{ steps.build.outputs.tests_total }}
-          tests-passed: ${{ steps.build.outputs.tests_passed }}
-          tests-failed: ${{ steps.build.outputs.tests_failed }}
-          tests-skipped: ${{ steps.build.outputs.tests_skipped }}
+          tests-total: ${{ steps.tests.outputs.tests_total }}
+          tests-passed: ${{ steps.tests.outputs.tests_passed }}
+          tests-failed: ${{ steps.tests.outputs.tests_failed }}
+          tests-skipped: ${{ steps.tests.outputs.tests_skipped }}

Halbaroth commented:
Actually, `dune runtest` and `dune build @runtest` should done the same thing. The issue comes from the patch applied for camlp5, as always :D. Because of this patch, targets of the Makefile are not composable. In particular, the `build` target removes the patch, which is required by `dune build`. I modified the Makefile to ensure that the patch is correctly applied and removed for test targets.

Halbaroth commented:
@hgouraud PR ready for review ;)

a2line commented:
If `dune runtest` and `dune build @runtest` does the same, then the Windows CI logs I gave for [this commit](https://github.com/a2line/geneweb/commit/3e29843868a7e1ceb7a4ca84ef82a0c8dda339fc) passed for Windows without using `call_unpatch`, so I think the problem was more adding “build” as prerequisite for ci target.

Halbaroth commented:
> so I think the problem was more adding “build” as prerequisite for ci target.

It is precisely what I said. I added `call_unpatch` for sake of robustness.

a2line commented:
Can you please adapt the four calls in the `collect-metrics` step? (I gave you the diff. above.)

Halbaroth commented:
Sure, I forgot to do it!
